### ISSUE TITLE
[suggestion] support YAML as well

### ISSUE THREAD

**Issue by iMartyn** [tomnomnom/gron#57]
Given that YAML is a superset of JSON, and often things that are described in YAML are completely translatable to JSON (I'm thinking specifically Kubernetes manifests but there are others), a `-y` option to parse yaml (especially useful if it worked with ungron) would be kinda nifty.

Of course that adds to #35 complications and #23 #28 as yaml spec specifically specifies multiple YAML docs per file, but anyway, just a thought.

**Comment by digitalsanctum** [comment]
As a workaround, I've found [yq](https://mikefarah.gitbook.io/yq/) to be helpful in certain cases. 

You can use `yq r - -j` to read stdin and convert to JSON before piping to `gron`

**Comment by iMartyn** [comment]
Yeah, that's what I'm doing myself, but given it's very easy to add yaml support (there's even a PR open now), saving a pipe in the complex one-liners is a good thing(tm) in my book :-)

**Comment by jarda-wien** [comment]
It seems that [yq](https://mikefarah.gitbook.io/yq/usage/properties) is capable of handling gron's duties with the `-o=p` parameter. Also, it support YAML out of the box.

**Comment by jarda-wien** [comment]
> It seems that [yq](https://mikefarah.gitbook.io/yq/usage/properties) is capable of handling gron's duties with the `-o=p` parameter. Also, it support YAML out of the box.

Example from gron's README:
```bash
curl -s "https://api.github.com/repos/tomnomnom/gron/commits?per_page=1" | ./yq_linux_amd64 eval -o=p | grep commit.author
0.commit.author.name = Tom Hudson
0.commit.author.email = mail@tomnomnom.com
0.commit.author.date = 2020-11-07T01:17:42Z
```

**Comment by benatkin** [comment]
How would you show the comments? Just omit them? I think it would be cool to try to preserve them. The comments on the same line on the same line, the comments above a node above a node I figure, and any comments at the end of the file at the end of the file.

Plus there's JSON with comments, a format which I'm opposed to, but which sees a lot of use by M$.
